system/classes/KO7/REST/Format/JSON.php

```diff
@@ -0,0 +1,58 @@
+<?php
+/**
+ * Class for formatting REST-Response bodies as JSON
+ *
+ * @package        KO7\REST
+ *
+ * @copyright  (c) since 2016 Koseven Team
+ * @license        https://koseven.ga/LICENSE
+ */
+class KO7_REST_Format_JSON extends REST_Format {
+
+    /**
+     * Format function
+     *
+     * @throws REST_Exception
+     *
+     * @return string
+     */
+    public function format() : string
+    {
+        try
+        {
+            $body = json_encode($this->_body,
+                JSON_THROW_ON_ERROR
+                |JSON_FORCE_OBJECT
+                |JSON_NUMERIC_CHECK
+                |JSON_PRESERVE_ZERO_FRACTION
+                |JSON_UNESCAPED_UNICODE
+            );
+        }
+        catch(Exception $e)
+        {
+            $message = 'Unknown JSON error.';
+
+            switch(json_last_error()) {
```

Comment:
`json_last_error()` doesnt work at `JSON_THROW_ON_ERROR` replace to 
`throw new REST_Exception($e->getMessage(), NULL, $e->getCode(), $e);`

Reply:
Yeah you are right i misread that....i shouldn't work late night 😄 ...fixed it